app/src/main/java/studyBuddy/Session.java

```diff
@@ -0,0 +1,133 @@
+package studyBuddy;
+
+import java.sql.Timestamp;
+
+public class Session {
+    private Timestamp startTime;
+    private Timestamp endTime;
+    private String name;
+    // In minutes
+    private double expectedTime;
+    private double productiveTime;
+    private double totalTime;
+    // Expected value between 0 and 1
+    private double percentProductive;
+    // TODO: fill this in correctly according to implementation of Time Management module
+    // private TimeManagement timeManager;
+    private boolean sessionOngoing;
+
+    /**
+     * Constructs a new studyBuddy.Session
+     * Parameters are initialized to somewhat meaningless values
+     * Session needs to be started by user
+     */
+    public Session() {
+        startTime = null;
+        endTime = null;
+        name = null;
+        expectedTime = 0.0;
+        productiveTime = 0.0;
+        totalTime = 0.0;
+        // assume they were productive the whole time
+        percentProductive = 1.0;
+        expectedTime = 0.0;
+        // timeManager = null;
+        sessionOngoing = false;
+    }
+
+//    TODO
+//    public void setTimeManager(TimeManagement manager);
+
+    /**
+     * Starts a session, triggered by user interaction
+     * Sets the session name and expected duration as given by user
+     * Records the real time session start time
+     * @param sessionName the name of the task for this session
+     * @param expectedSessionTime the expected duration of this session /
+     *                            expected time to complete task
+     */
+    public void startSession(String sessionName, double expectedSessionTime) {
+        long currentTime = System.currentTimeMillis();
+        startTime = new Timestamp(currentTime);
+        name = sessionName;
+        expectedTime = expectedSessionTime;
+        sessionOngoing = true;
```

Comment:
Can you show me what that would look like?

Reply:
wherever the timer management is done, either in this class or in the timemanagement class, we'll have to update the view once every second, so whenever we do that we'd check whether the current time is equal to the end time and if so call the callback. if we went with the former approach it could just be a field of some abstract/interface type and the user would construct an anonymous class, with the latter we might have to add a field associated with a given end time and check an array of these classes